NetappReplication/NetappReplicationDriverPoC.md

```diff
@@ -0,0 +1,159 @@
+# Volume based replication with SODA Netapp driver PoC
+
+
+# Introduction
+
+This provides a basic integration of netapp driver for replication and SODA. SODA already supports host and array based replication for different storage vendors(ex: Huawei, DRBD, ScutechCMS etc.). This POC enable to enhance the list of drivers to support replication basic feature.
+
+The drivers to support replication and other features are available in SODA Dock repo.
+
+###  Github:
+SODA : [https://github.com/sodafoundation](https://github.com/sodafoundation)
+
+SODA Dock: https://github.com/sodafoundation/dock
+
+
+### Install SODA with Netapp Driver.
+
+To test and use this POC, SODA should be installed with netapp basic driver. Please follow the below steps:
+
+1. Clone all the three repo(api controller and dock):
+
+```
+git clone https://github.com/sodafoundation/api.git
+git clone https://github.com/sodafoundation/controller.git
+git clone https://github.com/sodafoundation/dock.git
+```
+
+2. Add the driver codes in dock repo:
+
+ $GOPATH/dock/contrib/drivers/netapp/ontap/dock/contrib/drivers/netapp/ontap/replication.go
+
+ $GOPATH/dock/contrib/drivers/netapp/ontap/dock/contrib/drivers/netapp/ontap/cli.go
+
+
+ 3. Modify replication_driver.go file and add for netapp driver in dock repo.
+
+ vim contrib/drivers/replication_drivers.go
+
+ a. add import path
+ ```
+ import (
+         "github.com/sodafoundation/dock/contrib/drivers/netapp/ontap"
+)
+```
+b. NetappDRBDDriverType drivertype
+```
+ func InitReplicationDriver(resourceType string) (ReplicationDriver, error) {
+        var d ReplicationDriver
+        switch resourceType {
+       case driversConfig.NetappDRBDDriverType:
+               d = &ontap.ReplicationDriver{}
+               break
+        case driversConfig.DRBDDriverType:
+                d = &drbd.ReplicationDriver{}
+                break
+```
+
+4. Add the constansts variable in dock repo
+
+  vim contrib/drivers/netapp/ontap/constants.go
+```
+  const (
+     StorageDriverName = "netapp_ontap_san"
+  )
+```
+
+5. Install the soda using devsds script
+
+ a. run the ./install/devsds/install.sh from api replication
+
+ ```
+ Example:
+ root@root1-Latitude-E7450:~/gopath/src/github.com/sodafoundation/api# ./install/devsds/install.sh
+ ```
+
+
+6. Now manually netapp driver can be added.
+
+ a. kill the running osdsdock process
+  ```
+  kill -9 <dock process id>
+  ```
+
+7. Modify the opensds.conf file to add netapp_ontap_san driver.
+vi /etc/opensds/opensds.conf
+```
+[netapp_ontap_san]
+name = netapp_ontap_san
+description = netapp_ontap_san
+driver_name = netapp_ontap_san
+config_path = /etc/opensds/driver/netapp_ontap_san.yaml
+```
+```
+enabled_backends = lvm,nfs,netapp_ontap_san
+```
+
+8. Add the netapp_ontap_san.yml to /ete/opensds/driver directory
+
+ netapp_ontap_san.yml
+
+ ```
```

Comment:
Please mention clearly that this PoC is for replication of volumes from same aggregate on a SVM

Reply:
Done. Resolved all comments.